Patchers/PlayerDataPatcher/PatchedMethodCache.cs

```diff
@@ -0,0 +1,112 @@
+﻿using BepInEx;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+namespace SilksongPrepatcher.Patchers.PlayerDataPatcher;
+
+public class PatchedMethodCache
+{
+    // Dictionary [typeRef.FullName] -> List<methodRef.FullName>
+    public Dictionary<string, List<string>> PatchedMethods { get; set; } = new();
+
+    public void Add(string typeName, string methodName)
+    {
+        if (!PatchedMethods.TryGetValue(typeName, out List<string> methods)) methods = PatchedMethods[typeName] = new();
+        methods.Add(methodName);
+    }
+
+    public static string GetMetadataString()
+    {
+        string thisAssemblyVersion = typeof(PatchedMethodCache).Assembly.GetName().Version.ToString();
+
+        string assemblyCsharpPath = Path.Combine(Paths.ManagedPath, AssemblyNames.Assembly_CSharp);
+        DateTime assemblyCSharpModTime = File.GetLastWriteTimeUtc(assemblyCsharpPath);
+
+        return $"{thisAssemblyVersion} // {assemblyCSharpModTime}";
+    }
+
+    // Annoyingly there isn't any JSON available by default
+    public void Serialize(string filePath)
+    {
+        StringBuilder sb = new();
+
+        sb.AppendLine($"X {GetMetadataString()}");
+
+        foreach ((string typeName, List<string> methods) in PatchedMethods)
+        {
+            sb.AppendLine($"T {typeName}");
+            foreach (string methodName in methods)
+            {
+                sb.AppendLine($"M {methodName}");
+            }
+            sb.AppendLine($"E ");
+        }
+
+        File.WriteAllText(filePath, sb.ToString());
+    }
+
+    public static PatchedMethodCache? Deserialize(string filePath)
+    {
```

Comment:
IIRC there is some kind of logging available to preloader plugins and it would be nice if we could use it so it'll be less of a nightmare when this starts mysteriously returning null at unexpected times